### PR TITLE
Remove legacy SQL contacts implementation

### PR DESCRIPTION
**Pull request by lukasgraf**
This removes the SQL contacts implementation that was used for StABS.

Removed:
- All SQL Models
- Views, Forms, Resources (JS, CSS/SASS)
- Traversal Wrappers
- TabbedView tabs and listing views
- Custom participations implementation and storage in SQL
- Manual journal entry customizations
- SQLObjectSyncer
- Sender/recipient support in 'document from template' form in old UI
- SQL ContactsSource
- Registry Interfaces
- Testing feature, builders

Upgrade-Steps:
- Remove IContactSettings from registry
- Drop all SQL contacts related SQL tables
- Remove SQL contacts related permissions and update workflows

---

What I didn't remove yet are [these DocPropertyProviders](https://github.com/4teamwork/opengever.core/blob/88a51552347bda8924d5db550593a41a2f9e997c/opengever/contact/docprops.py) and the [adapters that they use](https://github.com/4teamwork/opengever.core/blob/88a51552347bda8924d5db550593a41a2f9e997c/opengever/contact/ogdsuser.py), especially the `OgdsUserToContactAdapter`.

I believe if these were just removed, we would break the creation of docproperties for OGDS users if they are selected as recipient/sender in the "document from template form" (new UI) when KuB is enabled. So this is more of a refactoring that needs to be looked at in the context of untangling the DocPropertiesProviders, and not straight forward code removal. Also doing that would have slowed me down, so I will create a follow up story for this, or at least a separate PR.

For [CA-2625](https://4teamwork.atlassian.net/browse/CA-2625)

## Checklist

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

- Upgrade-Steps:
  - [x] SQL Operations do not use imported model 
  - [x] Make it deferrable if possible _(not needed IMHO, dropping SQL tables is quick)_
  - [x] Execute as much as possible conditionally
  - DB-Schema migration
    - [x] All changes on a model (columns, etc) are included in a DB-schema migration.

[CA-2625]: https://4teamwork.atlassian.net/browse/CA-2625?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: opengever/core/upgrades/20230105125106_drop_sql_contacts_tables/upgrade.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    All SQL contacts data should have been migrated by previous migrations.
+    Therefore we can assert that the tables being dropped are empty.
+    """
+
+    def migrate(self):
+        portal = api.portal.get()
+        if portal.id != 'stabs':
+            # This upgrade only applies to StABS
+            return
```
Comment (njohner) [inline opengever/core/upgrades/20230105125106_drop_sql_contacts_tables/upgrade.py:45]: Are the tables created conditionally when setting up Gever?

**Review by njohner — APPROVED** [review]
Very nice and clean, thanks! You made it really easy to review (although I cannot say I reviewed in depth...)!

So I had just a few comments, otherwise LGTM!
Also did you check what needs to be done in gever-ui?

File: opengever/journal/form.py [hunk 2]
```diff
@@ -95,7 +73,8 @@ def updateActions(self):
         self.actions['add'].addClass("context")
 
     def createAndAdd(self, data):
-        contacts, users = self.split_contacts_and_users(data.get('contacts'))
+        contacts = []
+        users = []
         JournalManager(self.context).add_manual_entry(
```
[the remark this answers is not in the document]
Comment (lukasgraf) [reply opengever/journal/form.py:76]: Indeed, thank you. Fixed 👍

File: opengever/dossier/templatefolder/form.py [hunk 3]
```diff
@@ -186,9 +144,6 @@ class SelectTemplateDocumentWizardStep(
 
     def updateFieldsFromSchemata(self):
         super(SelectTemplateDocumentWizardStep, self).updateFieldsFromSchemata()
-        if not is_contact_feature_enabled():
```
[the remark this answers is not in the document]
Comment (lukasgraf) [reply opengever/dossier/templatefolder/form.py:189]: Indeed, thank you! I also was able to remove some more wizard-step related cruft, and made the form just a plain form, not a wizard any more 👍

File: opengever/core/upgrades/20230105125106_drop_sql_contacts_tables/upgrade.py (hunk 1 above)
Comment (lukasgraf) [reply opengever/core/upgrades/20230105125106_drop_sql_contacts_tables/upgrade.py]: Good catch, thanks! They are not, so I removed the conditional 👍

**Pull request merged**